### ISSUE TITLE
Issue with setting up node environment with proxy on macOS

### ISSUE THREAD

**Issue by Echomo-Xinyu** [reposense/RepoSense#1847]
Hi there, I am currently following the [set up guide](https://reposense.org/RepoSense/dg/settingUp.html) and have issues when running `./gradle clean build` command to verify the setup.

**Background**:
I am currently in mainland China and have to use a proxy for a better network connection.

**Attempts to fix the issue**:
I have tried to run the command with and without a proxy and neither works smoothly. 

- Running the command without a proxy is neglected as this relates to the network quality issue and there is not really much we can do about it.

The error log file for the attempt to install with proxy is attached here: 
[2022-12-14T05_22_01_617Z-debug.log](https://github.com/reposense/RepoSense/files/10224664/2022-12-14T05_22_01_617Z-debug.log)

The screenshot is here: 
![image](https://user-images.githubusercontent.com/25452497/207515294-12354677-8237-4d04-9452-d5864624434f.png)

- As the error message suggests, I should set up the 'proxy' config properly. But after I read the 'npm help config', the proxy is not really mentioned and we seem to be figuring out the set-up by ourselves. 
- I have then tried to search around and found this post: [maxos - Where to permanently set npm proxy?](https://stackoverflow.com/questions/39343263/where-to-permanently-set-npm-proxy), the content does not matter but the temporary fix method seems the solution I am looking for. The question uses "http://proxy.pro.com:8080" for the proxy address and port but this information is not available for me. 
- I have then searched about this and found this quora post [How-do-I-find-a-proxy-server-on-a-Mac](https://www.quora.com/How-do-I-find-a-proxy-server-on-a-Mac). I went to the corresponding location and found as screenshot below. I hence run `npm config set proxy 127.0.0.1:7890` and `npm config set https-proxy 127.0.0.1:7890` in shell. But when I run `./gradle clean build` again, it will result similar error which I have no clue to fix. *(there is additional error on missing react in the log file but I think that part can be resolved easily when the proxy is set properly so I decide not to put it here.)*

The screenshot of the error is provided here:

![image](https://user-images.githubusercontent.com/25452497/207904691-f97671e7-6f69-4f8b-a5ed-cb566c5f298c.png)

*ps: From my sheer knowledge of network, I know 127.0.0.1 is a local network address and hence it doesn't make sense to have the proxy server address as this. But I really do not know where to go in order to find such information.*

Can someone with similar experience share about how you have configured the npm with a proxy?

This is the end of my attempt to configure npm to download dependencies with proxy so far. Thank you so much for taking your time to read my post and I would really appreciate if you can provide me some guidance about how I may proceed from my current situation.

**Comment by parth-io** [comment]
Did you solve this issue?

If you try again without a proxy, you can try changing the npm registries to a Chinese one and then running Gradle again: https://stackoverflow.com/questions/22764407/npm-install-goes-to-dead-in-china

But if you're using a proxy, you should use the proxy server's address itself instead of 127.0.0.1 - the proxy provider would have its own address and port that you can find from Mac settings. The quora post you linked to shows how you can find it via MacOS's UI, or there's also a bash command `scutil --proxy` (from [here](https://superuser.com/questions/48480/how-can-i-get-mac-os-xs-proxy-information-in-a-bash-script))

Also, just a quick note, it's possible which `.npmrc` file the Gradle script uses, so it's possible that `npm config set proxy` may not have an effect on the `.npmrc` file used by the `npm` binary that Gradle itself launches . Perhaps doing `npm config set -g` to set the proxy setting in the global `.npmrc` may work better. The `.npmrc` file for RepoSense is at `<path to RepoSense>/build/node/lib/node_modules/npm/.npmrc` for me.

**Comment by Echomo-Xinyu** [comment]
Hi Parth, thank you for your reply and it looks very helpful! I will try the first method to see whether this helps.

Meanwhile, my apology for missing out the screenshot of my network setting on Mac earlier 
<img width="654" alt="image" src="https://user-images.githubusercontent.com/25452497/208232258-7b5683f1-5ea5-41fd-acdf-8a3bfc00574a.png">

As you may see, the place is pre-occupied by the "127.0.0.1" which makes no sense to me. I have also tried the bash command method but the output is as follows, which is no different from the UI side :( 
<img width="558" alt="image" src="https://user-images.githubusercontent.com/25452497/208232458-a6b8b59a-b4da-46d2-b541-760e4348b6c7.png">

**Comment by parth-io** [comment]
> have to use a proxy for a better network connection

Hmm, so when you say you're using a proxy, how do you connect to that proxy? There must be some way to get the proxy address, like from when you first set up the proxy connection

Yea, I don't get why you have localhost or 127.0.0.1 as a proxy either

**Comment by Echomo-Xinyu** [comment]
> Hmm, so when you say you're using a proxy, how do you connect to that proxy?

I am still waiting for the reply from the proxy provider. I'm using ClashX and am not very sure about the technical details. But all I did is to paste a given URL into the configuration of ClashX and a list of servers are then loaded for me to choose from. I don't think the URL I pasted is the proxy address also (after trying). So I guess I shall wait for the reply from the proxy provider.

Besides, I have tried to set up the environment again following the advice of using the Chinese mirror of npm but still encounter a network issue just as before. This is probably due to the bad network of my home :< So I guess I probably will postpone the issue when I'm back in Singapore. But still, a really big thank you for your suggestion!!!
